### PR TITLE
Better handle getting new markers

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#44

### CODE REVIEW

File: app/scripts/controllers/addMarker.js [hunk 1]
```diff
@@ -102,17 +102,17 @@ angular.module('publicEducationApp')
...
 
-      Marker.addMarker(venue, $scope.text, $scope.file, location, $scope.user).then(function() {
-        // After resolve promises Marker.addMarker.
-      });
+      // Getting the promise of add a new marker
+      var addingMarker = Marker.addMarker(venue, $scope.text, $scope.file, location, $scope.user);
 
-      $scope.setState('completed');
+      addingMarker.then(function() {
+        $scope.setState('completed');
```
Comment (ceoaliongroo) [inline app/scripts/controllers/addMarker.js]: Remove from `then`.

File: app/scripts/controllers/main.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 angular.module('publicEducationApp')
-  .controller('ListMarkersCtrl', function ($scope, Leaflet, storage, Marker, $location, $timeout) {
+  .controller('ListMarkersCtrl', function ($scope, Leaflet, storage, Marker, $location, $timeout, $rootScope) {
```
Comment (ceoaliongroo) [inline app/scripts/controllers/main.js]: remove `$rootscope`

File: app/scripts/controllers/main.js [hunk 3]
```diff
@@ -14,11 +14,14 @@ angular.module('publicEducationApp')
...
     $scope.markers = {};
 
+
     // Get markers.
     var getMarkers = function() {
       Marker.gettingMarkers().then(function(data) {
-
-        data = data.data;
+       if (data === {}) {
+          // Rare case, mostly in development, were all markers have been deleted.
```
Comment (amitaibu) [inline app/scripts/controllers/main.js]: Fix indentation.

File: app/scripts/controllers/playMarker.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ angular.module('publicEducationApp')
     $scope.playList = [];
     $scope.currentTrack = 0;
 
+    console.log('PlayMakerCtrl');
```
Comment (amitaibu) [inline app/scripts/controllers/playMarker.js]: Remove.

File: app/scripts/controllers/playMarker.js [hunk 5]
```diff
@@ -27,7 +28,7 @@ angular.module('publicEducationApp')
       $scope.selectedMarker = $scope.markers[$scope.venueId];
       angular.forEach($scope.selectedMarker.playList, function(value) {
         // Push the new items to the play list.
-        $scope.playList.push(value);
+          $scope.playList.push(value);
```
Comment (amitaibu) [inline app/scripts/controllers/playMarker.js]: Indentation.

File: app/scripts/services/marker.js [hunk 6]
```diff
@@ -1,13 +1,14 @@
 'use strict';
 
 angular.module('publicEducationApp')
-  .service('Marker', function Marker($q, $http, $timeout, BACKEND_URL, Phonegap) {
+  .service('Marker', function Marker($q, $http, $timeout, BACKEND_URL, Phonegap, $filter) {
```
Comment (amitaibu) [inline app/scripts/services/marker.js]: Remove `$filter`

File: app/scripts/services/marker.js [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
 
     return {
 
       // Private variable to hold the state.
       data: {
-        markers: null
+        markers: null,
+        lastProcessingHash: null
```
Comment (amitaibu) [inline app/scripts/services/marker.js:17]: Add doc about what it is.

File: app/scripts/services/marker.js [hunk 8]
```diff
@@ -91,12 +98,33 @@ angular.module('publicEducationApp')
...
+          if (self.isProcessing(data)) {
+            // Set timeout to abort cache in case of server issues.
+            $timeout(function() {
+              self.data.lastProcessingHash = null;
+            }, 600000);
+
+            defer.resolve(self.data.markers);
+          }
+          else {
+            // Update cache
```
Comment (amitaibu) [inline app/scripts/services/marker.js]: Finish with dot.

File: app/scripts/services/marker.js [hunk 9]
```diff
@@ -139,7 +167,7 @@ angular.module('publicEducationApp')
         options.params = {marker: JSON.stringify(marker)};
 
         ft.upload(fileURI, BACKEND_URL + '/add-marker', function onSuccess(result) {
-          console.log('Response = ' + result.response);
+          // console.log('Response = ', result);
```
Comment (amitaibu) [inline app/scripts/services/marker.js]: Remove.

File: app/scripts/services/marker.js [hunk 10]
```diff
@@ -148,6 +176,43 @@ angular.module('publicEducationApp')
...
+        return (self.data.lastProcessingHash) ? true : false;
+      },
+      /**
+       * Set the a status property to know when the marker was processed by the server.
+       *
+       * @param hash
+       *   Hash to identify the last marker inserted.
+       */
+      setProcessing: function(hash) {
+        if (hash) {
```
Comment (amitaibu) [inline app/scripts/services/marker.js]: Remove.

File: app/scripts/services/phonegap.js [hunk 11]
```diff
@@ -71,6 +69,8 @@ angular.module('publicEducationApp')
...
                 }
               }
+            }).success(function(data) {
+             return onSuccess(data);
```
Comment (amitaibu) [inline app/scripts/services/phonegap.js:73]: indentation.

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**